src/structures/Widget.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const Base = require('./Base');
+const WidgetUser = require('./WidgetUser');
+const WidgetChannelManager = require('../managers/WidgetChannelManager');
+
+/**
+ * Represents the data about the guild any bot can preview, connected to the specified guild.
+ * @extends {Base}
+ */
+class Widget extends Base {
+  constructor(client, data, id) {
+    super(client);
+
+    if (!data) return;
```

Comment:
Will a widget ever be constructed without data? Even if it will, you should make the if check call `_patch` at the end of the constructor